### PR TITLE
Disposable weapons Merge

### PR DESCRIPTION
**Pull request by LiamMillerSE**
Added Vulkan V64 Autocannons
Gave ability to grant players weapons after certain score threshold is reached
Guns now have ammo. When the current gun runs out of ammo, player goes back to their default gun
Bullets can now have conditions. Conditions are effects that can do things like damage over time to enemies

### CODE REVIEW

File: Assets/Scripts/Bullets/VulkanV64Trail.cs [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+    {
+        muzzleVelocity = 180;
+        mass = .1f; //The Mass controlls how slowed down the bike is by recoil
+        damageDealt = 1000;
+}
+
+
+public override void Shoot(Vector3 curPosition, Vector3 direction, Vector3 initialVelocity)
+    {
+        Debug.Log("VULKAN FIRE!");
```
Comment (toryjleo) [inline Assets/Scripts/Bullets/VulkanV64Trail.cs:19]: Please comment out Debug statements for master

File: Assets/Scripts/Bullets/VulkanV64Trail.cs [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+            if (hit.collider.gameObject.tag == "Enemy")
+            {
+                DealDamageAndDespawn(hit.collider.gameObject);
+            }
+        }
+        TrailRenderer trail = Instantiate(BulletTrail, curPosition, Quaternion.identity);
+        SpawnTrail(trail, shotRay.GetPoint(SHOT_RANGE));
+    }
+
+    private IEnumerator SpawnTrail(TrailRenderer trail, Vector3 endLoc)
```
Comment (toryjleo) [inline Assets/Scripts/Bullets/VulkanV64Trail.cs:40]: Please comment

File: Assets/Scripts/Conditions/Burning.cs [hunk 3]
```diff
@@ -0,0 +1,21 @@
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+public class Burning : Condition
```
Comment (toryjleo) [inline Assets/Scripts/Conditions/Burning.cs:5]: Please comment class summary

File: Assets/Scripts/Conditions/Condition.cs [hunk 4]
```diff
@@ -0,0 +1,26 @@
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+public abstract class Condition : MonoBehaviour
```
Comment (toryjleo) [inline Assets/Scripts/Conditions/Condition.cs:5]: Please comment class summary

File: Assets/Scripts/Conditions/Condition.cs [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+{
+    protected float tickRate = 2f;
+    private float lastInflicted = 0f;
+    protected Ai hostAi;
+
+    public void Tick()
```
Comment (toryjleo) [inline Assets/Scripts/Conditions/Condition.cs:11]: Please comment method summary

File: Assets/Scripts/Conditions/Condition.cs [hunk 6]
```diff
@@ -0,0 +1,26 @@
...
+    {
+        if (Time.time - lastInflicted > 1 / tickRate)
+        {
+            lastInflicted = Time.time;
+            InflictEffect();
+        }
+    }
+
+    public void SetHostAi(Ai host)
```
Comment (toryjleo) [inline Assets/Scripts/Conditions/Condition.cs:20]: Ditto

File: Assets/Scripts/Guns/GunCrate.cs [hunk 7]
```diff
@@ -0,0 +1,17 @@
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+public class GunCrate : MonoBehaviour
```
Comment (toryjleo) [inline Assets/Scripts/Guns/GunCrate.cs:5]: Ditto

File: Assets/Scripts/Guns/PlayerGuns/OctoBarrelLMG.cs [hunk 8]
```diff
@@ -0,0 +1,54 @@
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+
+public class OctoBarrelLMG : Gun
```
Comment (toryjleo) [inline Assets/Scripts/Guns/PlayerGuns/OctoBarrelLMG.cs:6]: Ditto Class Summary Comment

File: Assets/Scripts/Guns/PlayerGuns/VulcanV64AutoCannons.cs [hunk 9]
```diff
@@ -0,0 +1,60 @@
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+public class VulcanV64AutoCannons : Gun
```
Comment (toryjleo) [inline Assets/Scripts/Guns/PlayerGuns/VulcanV64AutoCannons.cs:5]: Ditto class summary comment

**Review by toryjleo — COMMENTED** [review]
Additionally I get an error when running the scene:

NullReferenceException: Object reference not set to an instance of an object
BikeScript.EquipGun (Gun gunToEquip) (at Assets/Scripts/Player/BikeScript.cs:133)
BikeScript.InitializeStartingGun () (at Assets/Scripts/Player/BikeScript.cs:145)
BikeScript.Awake () (at Assets/Scripts/Player/BikeScript.cs:76)

File: Assets/Scripts/Bullets/VulkanV64Trail.cs (hunk 1 above)
Comment (LiamMillerSE) [reply Assets/Scripts/Bullets/VulkanV64Trail.cs:19]: Commented out

File: Assets/Scripts/Guns/GunCrate.cs (hunk 7 above)
Comment (LiamMillerSE) [reply Assets/Scripts/Guns/GunCrate.cs:5]: GunCrate was not supposed to be in SC. Removed this file

**Review by toryjleo — CHANGES_REQUESTED** [review]
Get error when running:

ArgumentNullException: Value cannot be null.
Parameter name: source
UnityEngine.AudioSource.Play () (at <8e2857b79be4468ca1c28dda75978191>:0)
VulcanV64AutoCannons.Shoot (UnityEngine.Vector3 initialVelocity) (at Assets/Scripts/Guns/PlayerGuns/VulcanV64AutoCannons.cs:48)
BikeScript.FixedUpdate () (at Assets/Scripts/Player/BikeScript.cs:90)

**Review by toryjleo — APPROVED** [review]
Cool! LGTM!

**Pull request merged**